### DISCUSSION TITLE
Data fetching `pick` nested keys in Nuxt 3

### DISCUSSION THREAD

**Discussion by ol1s** [nuxt/nuxt#17760] [Ideas]
For the data fetching methods in Nuxt 3 (`useAsyncData`, `useFetch`, etc) there is an option called `pick` which allows us to define an array of keys from our data response that we want to keep in our page payload.

This is useful as it means we only pick the keys we actually want to use.

However, currently it seems that this array is limited to top-level keys only. It would be great if we could use something like dot-notation to also pick nested keys. For example:

```
pick: [
  "title",
  "hero.image",
],
```

**Comment by Rigo-m** [comment]
https://github.com/nuxt/framework/discussions/1407
